streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindows.java

```diff
@@ -52,6 +53,9 @@
  */
 public final class TimeWindows extends Windows<TimeWindow> {
 
+    private static final long DEFAULT_MAINTAIN_DURATION_MS = 24 * 60 * 60 * 1000L; // default: one day
```

Comment:
I can't think of a good reason for time- and join-windows to share this default, so they may as well just declare it separately. This is similar reasoning to why `SessionWindows` declares its default separately.

Note that `UnlimitedWindows`, which is also a subclass of `Windows`, has a different "default" retention period (`Long.MAX_VALUE`). Further evidence that it's actually not right to declare it in the abstract superclass.

Reply:
> This is similar reasoning to why SessionWindows declares its default separately.

Something I don't like -- I think we should use the same default for all windows (that need one) and thus share the default value as a single constant -- I would rather prefer to create an internal class that just define this single const and use for all window implementations.

I by the argument to not add it to `Windows` as `UnlimitedWindows` does not have a maintain duration. Thus, adding a new internal class for adding the constant that time/join/session window can just use without inheriting anything.